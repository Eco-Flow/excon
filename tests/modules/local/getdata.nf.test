nextflow_process {

    name "Test Process GET_DATA"
    script "modules/local/getdata.nf"
    process "GET_DATA"

    test("Should run without failures") {

        when {
            params {
                // define parameters here:
                clean = false
                outdir = "results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of( "metazoa_mart" )
                input[1] = Channel.of( "amellifera_eg_gene" )
                """
            }
        }

        then {
            //check if test case succeeded
            assert process.success
            //analyze trace file
            assert process.trace.tasks().size() == 1 

            //Assert snapshot of the two main output files
            assert snapshot(
              path("${launchDir}/${params.outdir}/go_files/background_gofiles/amellifera_eg_gene.go.txt"),
              //Cannot get a unique hit on the output fasta, probably because of the ordering of sequences.
              //path("${launchDir}/${params.outdir}/go_files/background_species/amellifera_eg_gene.fasta") 
            ).match("getdata") 
        }
    }
}
