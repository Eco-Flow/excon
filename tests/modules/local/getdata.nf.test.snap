{
    "getdata": {
        "content": [
            "amellifera_eg_gene.go.txt:md5,50fb9668e24ab2b3602327ca7b09cd1a"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-28T16:48:21.253066379"
    }
}